### ISSUE TITLE
[1.0.1] Doesn't wait for helper recomputation

### ISSUE THREAD

**Issue by boris-petrov** [emberjs/ember-test-helpers#469]
I have a helper in which I have something like:

```typescript
  public compute(): string {
    console.log('compute');
    return 'some text';
  }

  @observes('someService.foo')
  private onFoo(): void {
    console.log('recompute');
    this.recompute();
  }
```

And in the test I do:

```coffeescript
import { settled } from '@ember/test-helpers'
import wait from 'ember-test-helpers/wait'

...

    console.log 'setting 1'
    @owner.lookup('service:someService').set('foo', 'bar')
    console.log 'setting 2'

    await wait()
    await settled()
    console.log 'setting 3'

    assert.dom().hasText('some text')
```

When on `1.0.0` this test passes even without the `await settled()` line. `1.0.1` causes it to fail even with all of the above code. The output on `1.0.1` is:

```
setting 1
recompute
setting 2
setting 3
```

I'm kind of confused about all this waiting stuff but looking at `recompute`'s implementation I don't see anything concerning runloops so I'm not sure why it ever worked... and what I should do to make it pass on `1.0.1`?

**Comment by rwjblue** [comment]
Can you provide a reproduction repo? That would help me step through the code, and figure out whats going on...

**Comment by rwjblue** [comment]
Sounds basically like the issue in https://github.com/emberjs/ember.js/issues/14774.

**Comment by boris-petrov** [comment]
Yes, you are right, that fixes the issue. I wouldn't have thought about that, sorry. You're free to close the issue if you feel so.

**Comment by boris-petrov** [comment]
OK, so putting a `run` around `recompute` fixed the FE test but it broke an integration test of ours. Debugging it yields very strange results... we have something like:

```typescript
  @computed('someService.foo')
  public get a(): string {
    return this.someService.foo.split('-')[0];
  }

  @computed('someService.foo')
  public get b(): string {
    return this.someService.foo.split('-')[1];
  }

  public method(): void {
    console.log(this.a, this.b);
  }
```

And in some other place:

```typescript
    this.someService.set('foo', 'z-x');
    something.method();
```

The `console.log` in `method` only causes recomputation of one of `a` and `b` and not the other... the helper from the original post does call these methods but that actually breaks in a different place - i.e. it breaks other code. If I remove the `run` in the helper, this code starts working fine again. If I change the `run` to a `next` - it again works... no idea what's going on...

**Comment by rwjblue** [comment]
Just to be clear, it should be `run.join(this, this.recompute)`, that is what you are testing?

**Comment by boris-petrov** [comment]
Ah, no, I was using `run(() => this.recompute())`... your solution works in both cases. These things are really confusing... :D Thank you!

**Comment by boris-petrov** [comment]
This issue is fixed as of Ember 3.11.
